jetty-http/src/test/java/org/eclipse/jetty/http/matchers/HttpFieldsContainsHeaderValue.java

```diff
@@ -55,21 +55,7 @@ protected boolean matchesSafely(HttpFields fields)
         if (field == null)
             return false;
 
-        // Use HttpField.contains() logic
-        if (field.contains(this.value))
-            return true;
-
-        // Simple equals
-        if (this.value == field.getValue())
-            return true;
-
-        // Try individual value logic
-        String lcValue = this.value.toLowerCase(Locale.ENGLISH);
-        for (String value : field.getValues())
-        {
-            if (value.toLowerCase(Locale.ENGLISH).contains(lcValue))
-                return true;
-        }
-        return false;
```

Comment:
@joakime Do you recall why you implemented this?   values are not case insensitive and a string contains is not a good enough check.
For example if you want to check for "xyz", then it should match `xyz`, `foo,xyz,bar` and `"xyz"`, but not `XYZ` or `fooxyzbar`.

Reply:
This class was introduced in the junit5 revision - https://github.com/eclipse/jetty.project/commit/a3f1592c50c3af7001a5ab3c773768660eb8d07a
It is being used by multiple classes in that revision.

* [HpackDecoderTest](https://github.com/eclipse/jetty.project/blob/a3f1592c50c3af7001a5ab3c773768660eb8d07a/jetty-http2/http2-hpack/src/test/java/org/eclipse/jetty/http2/hpack/HpackDecoderTest.java)
* [HttpManyWaysToAsyncCommitTest](https://github.com/eclipse/jetty.project/blob/a3f1592c50c3af7001a5ab3c773768660eb8d07a/jetty-server/src/test/java/org/eclipse/jetty/server/HttpManyWaysToAsyncCommitTest.java)
* [DefaultServletTest](https://github.com/eclipse/jetty.project/blob/a3f1592c50c3af7001a5ab3c773768660eb8d07a/jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DefaultServletTest.java)
* [DefaultHandlerTest](https://github.com/eclipse/jetty.project/blob/a3f1592c50c3af7001a5ab3c773768660eb8d07a/tests/test-integration/src/test/java/org/eclipse/jetty/test/DefaultHandlerTest.java)

Lets take a look on the revision just before that junit5 commit and see what the test logic was/is ...

* [HpackDecoderTest](https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-http2/http2-hpack/src/test/java/org/eclipse/jetty/http2/hpack/HpackDecoderTest.java)
https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-http2/http2-hpack/src/test/java/org/eclipse/jetty/http2/hpack/HpackDecoderTest.java#L182
https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java#L173-L182
* [HttpManyWaysToAsyncCommitTest](https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-server/src/test/java/org/eclipse/jetty/server/HttpManyWaysToAsyncCommitTest.java)
https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-server/src/test/java/org/eclipse/jetty/server/HttpManyWaysToAsyncCommitTest.java#L334
https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-server/src/test/java/org/eclipse/jetty/server/AbstractHttpTest.java#L116-L119
* [DefaultServletTest](https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DefaultServletTest.java)
https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DefaultServletTest.java#L930-L934
https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-servlet/src/test/java/org/eclipse/jetty/servlet/DefaultServletTest.java#L1445-L1449
* [DefaultHandlerTest](https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/tests/test-integration/src/test/java/org/eclipse/jetty/test/DefaultHandlerTest.java)
https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/tests/test-integration/src/test/java/org/eclipse/jetty/test/DefaultHandlerTest.java#L173

Nothing obvious why it's case insensitive ATM.

I think it's because `HttpField.contains(String)` back then did that??

https://github.com/eclipse/jetty.project/blob/0ae0c500f0057d5c2f09e266bf8b15953a83ce22/jetty-http/src/main/java/org/eclipse/jetty/http/HttpField.java#L110

Reply:
Yeah, even our current `HttpField.contains(String)` does a the value check in a case insensitive way.

Reply:
I've looked at `HttpField.contains(String)` and I have my concerns about that as well. So I just made another Matcher for an exact match.

Reply:
> I've looked at `HttpField.contains(String)` and I have my concerns about that as well. So I just made another Matcher for an exact match.

We should open a new Issue to review that then.